--- freeradius-1.0.1.orig/src/modules/rlm_mschap/Makefile	2004-06-02 18:17:07.000000000 +0200
+++ freeradius-1.0.1/src/modules/rlm_mschap/Makefile	2004-10-13 09:04:16.171895090 +0200
@@ -14,7 +14,7 @@
 $(DYNAMIC_OBJS): $(HEADERS)
 
 smbencrypt: smbencrypt.o smbdes.o $(HEADERS)
-	$(CC) -I../../include $(LDFLAGS) -o smbencrypt smbencrypt.o smbdes.o ../../lib/libradius.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o smbencrypt ../../lib/md4.c smbencrypt.o smbdes.o
 
 smbencrypt-install:
 	$(INSTALL) -d -m 755 $(R)$(bindir)
