diff -dur freeradius-1.0.1.orig/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.in freeradius-1.0.1/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.in
--- freeradius-1.0.1.orig/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.in	2004-01-22 19:23:19.000000000 +0100
+++ freeradius-1.0.1/src/modules/rlm_sql/drivers/rlm_sql_mysql/configure.in	2004-09-30 14:24:40.165438814 +0200
@@ -32,7 +32,7 @@
 	then
 		sql_mysql_ldflags=`mysql_config --libs`
 		sql_mysql_cflags=`mysql_config --cflags`
-		AC_DEFINE(HAVE_MYSQL_H)
+		AC_DEFINE([HAVE_MYSQL_H], [], [Define if mysql.h is available])
 	else
 		AC_CHECK_LIB(z, compress, LIBS="$LIBS -lz")
 
@@ -55,7 +55,7 @@
 					MYSQL_INCLUDE=
 				)
 				if test "x$MYSQL_INCLUDE" != "x"; then
-					AC_DEFINE(HAVE_MYSQL_MYSQL_H)
+					AC_DEFINE([HAVE_MYSQL_H], [], [Define if mysql.h is available])
 					break;
 				fi
 			done
@@ -69,7 +69,7 @@
 		else
 			sql_mysql_cflags="${sql_mysql_cflags} ${MYSQL_INCLUDE}"
 			AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_MYSQL_H)
+			AC_DEFINE([HAVE_MYSQL_H], [], [Define if mysql.h is available])
   
 			AC_MSG_CHECKING([for mysql_init in -lmysqlclient])
 
