diff -Nur freeradius-1.1.6.orig/src/modules/rlm_eap/libeap/Makefile freeradius-1.1.6.linking/src/modules/rlm_eap/libeap/Makefile
--- freeradius-1.1.6.orig/src/modules/rlm_eap/libeap/Makefile	2006-03-23 11:17:46.000000000 +0000
+++ freeradius-1.1.6.linking/src/modules/rlm_eap/libeap/Makefile	2007-05-16 20:45:45.000000000 +0000
@@ -29,7 +29,7 @@
 $(TARGET).la: $(LT_OBJS)
 	$(LIBTOOL) --mode=link $(CC) -release $(RADIUSD_VERSION) \
 	$(LDFLAGS) $(LINK_MODE) -o $@ -rpath $(libdir) $^ \
-	$(top_builddir)/src/lib/libradius.la $(LIBS)
+	$(top_builddir)/src/lib/libradius.la $(LIBS) $(OPENSSL_LIBS)
 
 $(LT_OBJS): $(INCLUDES)
 
