--- freeradius-0.9.3/acinclude.m4.orig	2004-05-08 16:51:53.000000000 +0200
+++ freeradius-0.9.3/acinclude.m4	2004-05-08 17:06:49.355111483 +0200
@@ -157,12 +157,13 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#include <ucd-snmp/asn1.h>
-#include <ucd-snmp/snmp.h>
-#include <ucd-snmp/snmp_impl.h>],
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/library/asn1.h>
+#include <net-snmp/library/snmp.h>
+#include <net-snmp/library/snmp_impl.h>],
                [ int a = 1;],
-               SNMP_INCLUDE="";ucdsnmp=yes,
-               ucdsnmp=)
+               SNMP_INCLUDE="";netsnmp=yes,
+               netsnmp=)
 
 dnl #
 dnl #  If not, look for it in a number of directories and in ucd-snmp.
@@ -187,22 +188,23 @@
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
 #endif
-#include <ucd-snmp/asn1.h>
-#include <ucd-snmp/snmp.h>
-#include <ucd-snmp/snmp_impl.h>],
+#include <net-snmp/net-snmp-config.h>
+#include <net-snmp/library/asn1.h>
+#include <net-snmp/library/snmp.h>
+#include <net-snmp/library/snmp_impl.h>],
                    [ int a = 1;],
-                   SNMP_INCLUDE="-I$try";ucdsnmp=yes,
-                   ucdsnmp=)
-    if test "x$ucdsnmp" != "x"; then
+                   SNMP_INCLUDE="-I$try";netsnmp=yes,
+                   netsnmp=)
+    if test "x$netsnmp" != "x"; then
       break;
     fi
   done
   CFLAGS="$old_CFLAGS"
 fi
 
-if test "x$ucdsnmp" = "x"; then
+if test "x$netsnmp" = "x"; then
   old_CFLAGS="$CFLAGS"
-  for try in /usr/include/ucd-snmp /usr/local/include/ucd-snmp $snmp_include_dir; do
+  for try in /usr/include/net-snmp /usr/local/include/net-snmp $snmp_include_dir; do
     CFLAGS="$old_CFLAGS -I$try"
 dnl #
 dnl #  First, see if we can build it WITHOUT using any special includes and without ucd-snmp
@@ -227,9 +229,9 @@
 #include <snmp.h>
 #include <snmp_impl.h>],
                [ int a = 1;],
-               SNMP_INCLUDE="";ucdsnmp=no,
-               ucdsnmp=)
-    if test "x$ucdsnmp" != "x"; then
+               SNMP_INCLUDE="";netsnmp=no,
+               netsnmp=)
+    if test "x$netsnmp" != "x"; then
       break;
     fi
   done
@@ -239,9 +239,9 @@
 dnl #
 dnl #  If not, look for it in a number of directories and without ucd-snmp
 dnl #
-if test "x$ucdsnmp" = "x"; then
+if test "x$netsnmp" = "x"; then
   old_CFLAGS="$CFLAGS"
-  for try in /usr/include/ucd-snmp /usr/local/include/ucd-snmp $snmp_include_dir; do
+  for try in /usr/include/net-snmp /usr/local/include/net-snmp $snmp_include_dir; do
     CFLAGS="$old_CFLAGS -I$try"
     AC_TRY_COMPILE([
 #ifdef HAVE_SYS_TYPES_H
@@ -263,21 +263,21 @@
 #include <snmp.h>
 #include <snmp_impl.h>],
                    [ int a = 1;],
-                   SNMP_INCLUDE="-I$try";ucdsnmp=no,
-                   ucdsnmp=)
-    if test "x$ucdsnmp" != "x"; then
+                   SNMP_INCLUDE="-I$try";netsnmp=no,
+                   netsnmp=)
+    if test "x$netsnmp" != "x"; then
       break;
     fi
   done
   CFLAGS="$old_CFLAGS"
 fi
 
-if test "x$ucdsnmp" = "x"; then
+if test "x$netsnmp" = "x"; then
   AC_MSG_RESULT(no)
 else
-  if test "x$ucdsnmp" = "xyes"; then
-    AC_MSG_RESULT((ucd-snmp)yes)
-    AC_DEFINE(HAVE_UCD_SNMP_ASN1_SNMP_SNMPIMPL_H)
+  if test "x$netsnmp" = "xyes"; then
+    AC_MSG_RESULT((net-snmp)yes)
+    AC_DEFINE(HAVE_NET_SNMP_LIBRARY_ASN1_SNMP_SNMPIMPL_H)
   else
     AC_MSG_RESULT(yes)
     AC_DEFINE(HAVE_ASN1_SNMP_SNMPIMPL_H)
@@ -285,24 +285,24 @@
 dnl #
 dnl #  Now do the same thing, looking for the SNMP library directory
 dnl #
-  AC_MSG_CHECKING([for snmp_build_var_op in -lsnmp])
+  AC_MSG_CHECKING([for snmp_build_var_op in -lnetsnmp])
 
 dnl #
 dnl #  First, see if we can build it WITHOUT using any special includes
 dnl #
   old_LIBS="$LIBS"
-  LIBS="$old_LIBS -lsnmp"
+  LIBS="$old_LIBS -lnetsnmp"
   AC_TRY_LINK([extern char snmp_build_var_op();],
               [ snmp_build_var_op()],
-              SNMP_LIBS="-lsnmp",
+              SNMP_LIBS="-lnetsnmp",
               SNMP_LIBS=)
 
   if test "x$SNMP_LIBS" = "x"; then
     for try in /usr/lib /usr/local/lib /usr/local/snmp/lib $snmp_lib_dir; do
-      LIBS="$old_LIBS -L$try -lsnmp"
+      LIBS="$old_LIBS -L$try -lnetsnmp"
       AC_TRY_LINK([extern char snmp_build_var_op();],
                   [ snmp_build_var_op()],
-                  SNMP_LIBS="-L$try -lsnmp",
+                  SNMP_LIBS="-L$try -lnetsnmp",
                   SNMP_LIBS=)
       if test "x$SNMP_LIBS" != "x"; then
         break;
@@ -311,10 +311,10 @@
 dnl   #  That didn't work.  Try adding the '-lcrypto' line.
 dnl   #  Some SNMP libraries are linked against SSL...
 dnl   #
-      LIBS="$old_LIBS -L$try -lsnmp -lcrypto"
+      LIBS="$old_LIBS -L$try -lnetsnmp -lcrypto"
       AC_TRY_LINK([extern char snmp_build_var_op();],
                   [ snmp_build_var_op()],
-                  SNMP_LIBS="-L$try -lsnmp -lcrypto",
+                  SNMP_LIBS="-L$try -lnetsnmp -lcrypto",
                   SNMP_LIBS=)
       if test "x$SNMP_LIBS" != "x"; then
         break;
--- freeradius-0.9.3/src/include/radius_snmp.h.orig	2002-08-16 22:15:22.000000000 +0200
+++ freeradius-0.9.3/src/include/radius_snmp.h	2004-05-08 17:08:33.433163385 +0200
@@ -9,10 +9,11 @@
 #include	<asn1.h>
 #include	<snmp.h>
 #include	<snmp_impl.h>
-#elif HAVE_UCD_SNMP_ASN1_SNMP_SNMPIMPL_H
-#include	<ucd-snmp/asn1.h>
-#include	<ucd-snmp/snmp.h>
-#include	<ucd-snmp/snmp_impl.h>
+#elif HAVE_NET_SNMP_LIBRARY_ASN1_SNMP_SNMPIMPL_H
+#include	<net-snmp/net-snmp-config.h>
+#include	<net-snmp/library/asn1.h>
+#include	<net-snmp/library/snmp.h>
+#include	<net-snmp/library/snmp_impl.h>
 #endif
 
 #include        "smux.h"
--- freeradius-0.9.3/acconfig.h.orig	2003-06-22 12:04:21.000000000 +0200
+++ freeradius-0.9.3/acconfig.h	2004-05-08 17:11:04.883863944 +0200
@@ -69,8 +69,8 @@
 /* Include user collision code*/
 #undef WITH_USERCOLLIDE
 
-/* Define if you have the <ucd-snmp/asn1.h>, <ucd-snmp/snmp_impl.h> and <ucd-snmp/snmp.h> header file.  */
-#undef HAVE_UCD_SNMP_ASN1_SNMP_SNMPIMPL_H
+/* Define if you have the <net-snmp/library/asn1.h>, <net-snmp/library/snmp_impl.h> and <net-snmp/library/snmp.h> header file.  */
+#undef HAVE_NET_SNMP_LIBRARY_ASN1_SNMP_SNMPIMPL_H
 
 /* Define if you have the <asn1.h>, <snmp_impl.h> and <snmp.h> header file.  */
 #undef HAVE_ASN1_SNMP_SNMPIMPL_H
